### PR TITLE
Use Cucumber for E2E tests, add tests around Policy Playground and Policy CRUD operations

### LINKED ISSUE #50: E2E Tests
* Add additional Cypress tests to cover the following features
  * Policy Evaluation
  * Editing a policy
  * Deleting a policy
* Improve e2e testing experience
  * Abstract away some common steps
  * Figure out how to better integrate with the backend if running against a local instance of Rode
  * Are we seeding data in Rode? If yes, could update the mocked responses to reflect the seed data so we could run the existing test suite against the real Rode Demo instance.

* Create a Smoke Test group to cover basic functionality that we want to ensure
* Figure out how to run the subset of the tests in CI so we don't have to run the entire test suite every time

### PR DESCRIPTION
**Pull request by pickjasmine**
Closes #50 

* Adds Cucumber for integration tests
* Adds additional tests around Policy Playground and Policy CRUD operations
* Adds `yarn smoke` command to run the tests tagged with `@smoke` in the command line
* Removes extra buttons on the policy search page that are no longer needed with navigation updates
* Larger cards for vulnerabilities and builds when viewing resource details

For now, I'm going to keep all e2e tests running in this pipeline for the safety of making sure errors and expected interactions stay intact in the UI.

### CODE REVIEW

File: cypress/integration/Policy.feature [hunk 1]
```diff
@@ -0,0 +1,91 @@
+Feature: Policies
+
+  Scenario: Open the policy search page
+    Given I open the application
+    When I navigate to the "PolicySearch" page
+    Then I see "PolicySearchInput"
+
+  Scenario: Search for a non-existent policy
+    Given I am on the "PolicySearch" page
+    When I search for "NonExistent" policy
```
Comment (alexashley) [inline cypress/integration/Policy.feature:10]:
```suggestion
    When I search for a "NonExistent" policy
```
Nitpick, but this one and the line below could probably use an article

File: cypress/integration/Policy.feature [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    And I click the "SavePolicy" button
+    Then I no longer see "<message>" message
+    Scenarios:
+    | message | text | input |
+    | PolicyNameRequired | name | PolicyName |
+    | PolicyRegoRequired | rego | PolicyRegoContent |
+
+  Scenario Outline: Create policy - validating rego code
+    Given I am on the "CreatePolicy" page
+    When I test <validity> rego policy code
```
Comment (alexashley) [inline cypress/integration/Policy.feature:43]: Nitpick, but I think Rego is a proper noun
```suggestion
    When I test <validity> Rego policy code
```

File: cypress/integration/Policy/policy.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    mockFailedPolicyValidation
+  );
+
+  cy.get(selectors.PolicyRegoContentInput)
+    .clear()
+    .type(`Package ${chance.word()}`);
+  cy.get(selectors.ValidatePolicyButton).click();
+});
+
+When(/^I test valid rego policy code$/, () => {
```
Comment (alexashley) [inline cypress/integration/Policy/policy.js:64]: Thoughts on merging this and the previous step? I don't know the name of the feature, but there's a way to get the value of `valid/invalid` passed to the step by using `{string}` (or whatever type) where that work would go: https://github.com/TheBrainFamily/cypress-cucumber-preprocessor#reusable-step-definitions

Comment (pickjasmine) [reply cypress/integration/Policy/policy.js:64]: I agree with merging. I tried to combining the steps before, but I had some issues with mocking the request/response, but I restructured some things so let me try again.

Comment (pickjasmine) [reply cypress/integration/Policy/policy.js:64]: JK was actually just a problem with my RegEx 🤦  Thanks for the suggestions!

**Review by alexashley — APPROVED** [review]
:shipit:

**Pull request merged**